### DISCUSSION TITLE
Green Zones Model

### DISCUSSION THREAD

**Discussion by ColtMercer** [nautobot/nautobot#1709] [Ideas]
Should be able to create Green Zones for changes for business groups that specify what time windows are allowed for changes. Then associate those Green Zones to devices. These are used to automate change record creation in our current system we would like to replace with Nautobot.

**Comment by jathanism** [comment]
Sounds like you're talking about maintenance windows and/or severity levels for change events?

**Reply by ColtMercer** [reply]
Yes the Start and End Time of a maintenance window and the days associated.
Example:
Start                                  End
Saturday 05:00 (GMT)         Saturday 10:00 (GMT)
Sunday 05:00 (GMT)         Sunday 10:00 (GMT)

We use this field in our current system to schedule change records. Replacing with Nautobot and need this functionality

**Reply by ColtMercer** [reply]
To add to this we also have Green and Yellow zones. Green Zones we raise impacting changes. Yellow zones we can perform changes but with the intention of not impacting production

**Comment by alextremblay** [comment]
Sounds like something that could be implemented as a nautobot plugin, if there's enough interest for someone to write one
